packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -30,210 +31,214 @@ export type CreateViewResult = {
   view: VideoStreamRendererView;
 };
 
-async function createViewRemoteVideo(
+async function createViewVideo(
   context: CallContext,
   internalContext: InternalCallContext,
   callId: string,
-  participantId: CommunicationIdentifierKind | string,
-  stream: RemoteVideoStreamState,
+  stream?: RemoteVideoStreamState | LocalVideoStreamState,
+  participantId?: CommunicationIdentifierKind | string,
   options?: CreateViewOptions
 ): Promise<CreateViewResult | undefined> {
-  // Render RemoteVideoStream that is part of a Call
-  const streamId = stream.id;
-  const streamType = stream.mediaStreamType;
-  let participantKey;
-  if (typeof participantId === 'string') {
-    participantKey = participantId;
-  } else {
-    participantKey = toFlatCommunicationIdentifier(participantId);
-  }
-  const streamLogInfo = { callId, participantKey, streamId, streamType };
-
-  _logCreateRemoteStreamEvent(EventNames.CREATING_REMOTE_VIEW, streamLogInfo);
-  const renderInfo = internalContext.getRemoteRenderInfoForParticipant(callId, participantKey, streamId);
-
-  if (!renderInfo) {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_NOT_FOUND, streamLogInfo);
-    console.warn('RemoteVideoStream not found in state');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendered') {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_ALREADY_RENDERED, streamLogInfo);
-    console.warn('RemoteVideoStream is already rendered');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendering') {
-    // Do not log to console here as this is a very common situation due to UI rerenders while
-    // the video rendering is in progress.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_RENDERING, streamLogInfo);
-    return;
-  }
+  // we can only have 3 types of createView
+  let createViewType: 'local' | 'remote' | 'unparented';
 
-  // "Stopping" only happens if the stream was in "rendering" but `disposeView` was called.
-  // Now that `createView` has been re-called, we can flip the state back to "rendering".
-  if (renderInfo.status === 'Stopping') {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_STOPPING, streamLogInfo);
-    internalContext.setRemoteRenderInfo(
-      callId,
-      participantKey,
-      streamId,
-      renderInfo.stream,
-      'Rendering',
-      renderInfo.renderer
-    );
-    return;
-  }
-
-  const renderer = new VideoStreamRenderer(renderInfo.stream);
-
-  internalContext.setRemoteRenderInfo(callId, participantKey, streamId, renderInfo.stream, 'Rendering', undefined);
+  // we will reuse these for local as well but we need to make sure the remote stream is passed in like before.
+  let streamId;
```

Comment:
same here, these are only set once - if you can make these constant and set at the time of declaration